sklearn/linear_model/_ridge.py

```diff
@@ -235,6 +236,60 @@ def _solve_svd(X, y, alpha):
     return np.dot(Vt.T, d_UT_y).T
 
 
+def _solve_lbfgs(
+    X, y, alpha, positive=False, max_iter=None, tol=1e-3, X_offset=None, X_scale=None
+):
+    n_samples, n_features = X.shape
+
+    options = {}
+    if max_iter is not None:
+        options["maxiter"] = max_iter
+    if positive:
+        bounds = [(0, np.inf)] * n_features
+    else:
+        bounds = [(-np.inf, np.inf)] * n_features
```

Comment:
I agreed. It is very easy to rewrite code to accept lbfgs solver only when positive=True,
but I'm struggling to rewrite test cases because in most cases lbfgs results are
needed to compare with the results calculated using other solver.
Is it possible to use lbfgs even when positive=False?

Reply:
I fixed code to accept lbfgs solver only when `positive=True`.